src/app/features/courses/courses.component.ts

```diff
@@ -7,8 +7,14 @@ import {Course} from 'src/app/models/course';
 })
 export class CoursesComponent {
   courses: Course[];
-  @Input() courseBtnChildTitle = '';   
+  @Input() courseBtnChildTitle = '';
+  searchPlaceHolder = "Search course by Name";   
   editable: boolean = false;
+  getCourse(course: string) {
+    debugger;
+    var data = this.courses.filter(x => x.Name === course);
```

Comment:
use let/const instead of var
Also you can use spread operator to make your code cleaner 
`this.courses.filter(({ name }: Course) => name === course)`
add interface for course object